mongodbatlas/resource_mongodbatlas_cluster_test.go

```diff
@@ -1238,10 +1240,13 @@ func testAccMongoDBAtlasClusterConfigAWSWithAutoscaling(projectID, name, autoSca
 			provider_name               = "AWS"
 			provider_disk_iops 			    = 300
 			provider_encrypt_ebs_volume = false
-			provider_instance_size_name = "M30"
+			provider_instance_size_name = "%[8]s"
 			provider_region_name        = "EU_CENTRAL_1"
 			provider_auto_scaling_compute_min_instance_size = "%[5]s"
 			provider_auto_scaling_compute_max_instance_size = "%[6]s"
+			lifecycle{ // To simulate if there a new instance size name to avoid scale cluster down to original value
+				ignore_changes = [provider_instance_size_name]
```

Comment:
I don't think this quite tests what would happen in reality with auto scaling but it does test if upgraded via terraform and adds a lifecycle event. If using autoscaling terraform wouldn't likely to be the tool setting the larger provider_instance_size_name, Atlas would when a scale event is triggered.  You would need to test this manually, I believe, to truly test if lifecycle works as we assume, basically:
-create a cluster with at least autoscaling on and a max value with terraform, example: an M10 with a max of a M40
-go into the Atlas UI and change the cluster size to M20 to simulate an auto scale event
-run a plan and see if terraform sees the size changed
-destroy all that and do again but
-create a cluster with at least autoscaling on and a max with terraform, example: an M10 with a max of a M40 with added the lifecycle ignore_changes in the config
-go into the Atlas UI and change the cluster size to M20 to simulate an auto scale event
-run a plan and see if terraform sees the size changed or if it properly ignores it.   

I'm not sure this is an automatable test unless instead of using the UI you triggered the cluster to scale up with the API directly, outside of terraform.    To actually trigger a scaling event you'd have to create a sustained load over a period of time.

Reply:
I see, I did try using TF directly and website UI as you suggested and it seems worked for me, I mean, at first it detected changes from website UI using refresh and plan and the second time used the lifecycle and ommited the change using refresh then apply

Reply:
That makes sense - just wanted to make sure we did a test to best emulate this.  Ty!